programmers/이상한-문자-만들기.js

```diff
@@ -0,0 +1,20 @@
+function solution(s) {
+    const Arr = s.split(" ");
+    const result = [];
+    const wordArr = [];
+    
+    for (let i = 0; i < Arr.length; i += 1) {
+        wordArr.push(Arr[i].split(""));
+
+        for (let j = 0; j < Arr[i].length; j += 1) {
+            
+            j % 2 === 0 ? 
+                result.push(wordArr[i][j].toUpperCase())
+                :
+                result.push(wordArr[i][j].toLowerCase());
+        }
+        result.push(" ");
+    }
+    result.length = s.length;
+    return result.join("");
+}
```

Comment:
```suggestion
function solution(s) {
  const arr = s.split(' ');
  const result = [];
  const wordArr = [];

  for (let i = 0; i < arr.length; i += 1) {
    wordArr.push(arr[i].split(''));

    for (let j = 0; j < arr[i].length; j += 1) {
      j % 2
        ? result.push(wordArr[i][j].toLowerCase())
        : result.push(wordArr[i][j].toUpperCase());
    }
    result.push(' ');
  }
  return result.splice(0, result.length - 1).join('');
}
```
1. 변수명 만들때 카멜케이스 !
2. for문안에서 줄바꿈 문제있어여
3. j % 2 === 0 ? 순서1 : 순서2 ➡ j % 2 ? 순서2 : 순서1
4. length로 길이 자르는방법 아주 좋은데 splice메서드로 해결 가능합니다
5. for문을 map으로 변경할 수 있어요

```js
function solution(s) {
    return s.split(' ')
        .map(word => word.split('')
        .map((str, idx) => idx % 2 ? str.toLowerCase() : str.toUpperCase())
        .join('')).join(' ');
}
```
이건 김지수님꺼 코드인데 상당히 훌륭해서 가지고 왔습니다 ~